core/MessageID.h

```diff
@@ -2,6 +2,10 @@
 
 #include <cstdint>
 
+#ifdef ERROR
+    #undef ERROR
```

Comment:
Why is this needed?

Reply:
VS 2019 has some header already having ERROR defined. Generating compiling error without undef it.